### ISSUE TITLE
`torch.nn.functional.ctc_loss` inconsistent implementation and docs

### ISSUE THREAD

**Issue by zeshengzong** [pytorch/pytorch#150995]
### 📚 The doc issue

When fixing #150835 find out here's an inconsistency between doc and test of `torch.nn.functional.ctc_loss`.

The [doc](https://pytorch.org/docs/stable/generated/torch.nn.functional.ctc_loss.html) describe `targets` param `cannot be blank`

![Image](https://github.com/user-attachments/assets/f80abfe4-e716-408c-a735-91595569a057)

But there's a test case in `test_nn.py` do validate pass empty tensor as `targets` and compute loss result.

https://github.com/pytorch/pytorch/blob/4273e5d15cfcb282b2795684874ea439d8620999/test/test_nn.py#L11413-L11421

So it's confused whether `targets` is allowed to be blank or not.

### Suggest a potential alternative/fix

May change the doc description about `targets` param

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki
